### PR TITLE
Users can post articles. Users can also flag an article which is inap…

### PR DESCRIPTION
**Pull request by goodnessaig1**
…ppropriate for the community

### CODE REVIEW

File: server/controller/flaggedController.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const pool = require('../models/db');
+const queries = require('../queries/flagQuery');
+const moment = require('moment');
+
+
+
+class flagController {
+
+static async flagArticles(req,res){
```
Comment (courage173) [inline server/controller/flaggedController.js]: add this inside the article controller. you don't need this controller

File: server/controller/flaggedController.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            message:  'Article with this id not found',
+        });
+        }
+            if(!flagged){
+                return res.status(404).json({
+                    status: 'Error',
+                    message: 'Not provided'
+                })
+            }
+            const flaggedAt = moment().format('YYYY-MM-DD ');
```
Comment (courage173) [inline server/controller/flaggedController.js]: change to user luxon package instead of moment

File: server/models/articleModel.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const pool = require("./db");
+
+module.exports = () => {
+  const createCategoryTable = async () => {
```
Comment (courage173) [inline server/models/articleModel.js]: why is this named category?

File: server/test/article.spec.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+// ASSERTION STYLE
+chai.should();
+
+chai.use(chaiHttp);
+ let token;
+
+describe('POST ARTICLE ',()=>{
+    // LOGIN FIRST IN ORDER TO GET YOUR VALID TOKEN
+    describe('POST /auth/v1/login',()=>{
```
Comment (courage173) [inline server/test/article.spec.js:24]: why have each test in a different describe?. login a user in should be inside a before hook

File: server/test/article.spec.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            )
+            .end((err, response) =>{
+                response.should.have.status(201)
+                response.body.should.be.a('object');
+                response.body.should.have.property('status').eq('success');
+                done();
+            })
+        })
+        
+ afterEach(async () => {
```
Comment (courage173) [inline server/test/article.spec.js]: this should maybe be at the top close to the before hook

File: server/test/flag.spec.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+let chai = require('chai');
+let chaiHttp = require('chai-http');
```
Comment (courage173) [inline server/test/flag.spec.js]: everything in this file should be part of the article test

File: server/test/gifImg.spec.js [hunk 7]
```diff
@@ -38,7 +38,7 @@ describe('GIF UPLOAD IMAGE TO CLOUDINARY ',()=>{
         it('It should post an image to cloudinaary and also return the link', (done)=>{
             chai.request(server)
             .post('/v1/gifs')
-            .set("authorization", "Bearer " + token)
```
Comment (courage173) [inline server/test/gifImg.spec.js:41]: why did you remove it from being a bearer token???

**Review by courage173 — CHANGES_REQUESTED** [review]

File: server/controller/flaggedController.js (hunk 1 above)
Comment (goodnessaig1) [reply server/controller/flaggedController.js]: Ok sir

File: server/models/articleModel.js (hunk 3 above)
Comment (goodnessaig1) [reply server/models/articleModel.js]: It was a mistake sir. I have changed it now

File: server/test/flag.spec.js (hunk 6 above)
Comment (goodnessaig1) [reply server/test/flag.spec.js]: ok sir. i would move them now

File: server/test/gifImg.spec.js (hunk 7 above)
Comment (goodnessaig1) [reply server/test/gifImg.spec.js:41]: i was formerly having issues with the header. So when i changed to token, the test passed

File: server/test/article.spec.js (hunk 5 above)
Comment (goodnessaig1) [reply server/test/article.spec.js]: Ok sir.
Done

File: server/.env [hunk 8]
```diff
@@ -1,4 +1,5 @@
 SUPERSEC= password123
 CLOUD_NAME=e-commerce-website
 CLOUD_API_KEY=742572971787234
-CLOUD_API_SECRET=4ytWgJj4DLzBDJNh2_J1kL2mpUU
\ No newline at end of file
+CLOUD_API_SECRET=4ytWgJj4DLzBDJNh2_J1kL2mpUU
```
Comment (courage173) [inline server/.env]: why is your .env here??

File: server/.gitignore [hunk 9]
```diff
@@ -1 +1,2 @@
-/node_modules
\ No newline at end of file
+/node_modules
```
Comment (courage173) [inline server/.gitignore]: why did you remove .env from gitignore

File: server/controller/articleController.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+const getTime = () => {
+	const easternTime = DateTime.local();
+	return easternTime.toLocaleString(DateTime.DATETIME_FULL);
+};
+
+class ArticleController {
+  static async createArticle(req, res) {
+      try {
+          const { title, article, categoryId } = req.body;
+          let {flagged} = req.body
```
Comment (courage173) [inline server/controller/articleController.js]: why are you passing flagged when creating an article?? i don't understand

Comment (courage173) [reply server/controller/articleController.js]: you already did it in the flagArticle why are you doing it here again

File: server/controller/articleController.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+      return res.status(404).json({
+        status: 'Failed',
+        message: 'Category with the specified categoryId NOT found',
+      });
+    }
+     if(!flagged){
+        flagged = false
+    }
+
+    const createdAt = moment().format('YYYY-MM-DD ');
```
Comment (courage173) [inline server/controller/articleController.js]: why is there extra space after the DD?

File: server/controller/articleController.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+  static async flagArticles(req,res){
+       try {
+            const {articleId} = req.params
+            const {flagged} = req.body
+
+            const article = await pool.query(queries.selectArticle, [articleId]);
+
+            if (article.rows.length === 0) {
+        return res.status(404).json({
```
Comment (courage173) [inline server/controller/articleController.js]: where is your eslint and prettier, please lint this properly

File: server/controller/articleController.js [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+            status: 'Failed',
+            message:  'Article with this id not found',
+        });
+        }
+            if(!flagged){
```
Comment (courage173) [inline server/controller/articleController.js:62]: this should be validated with a middleware

File: server/controller/articleController.js (hunk 11 above)
Comment (courage173) [inline server/controller/articleController.js]: change anywhere you are using moment to luxon DateTime

File: server/test/gifImg.spec.js (hunk 7 above)
Comment (courage173) [reply server/test/gifImg.spec.js:41]: please change it back and debug why you are getting the issue

Comment (goodnessaig1) [reply server/test/gifImg.spec.js:41]: The issue is that, if i do it  this way .set("authorization", "Bearer " + token), it would give me a 401 response which is as a result of an invalid token provided. 
But when i do it this way, 
.set('token', token)
it gives me a 201 response which is OK

File: server/.env (hunk 8 above)
Comment (goodnessaig1) [reply server/.env]: i did not add it well in my gitignore. I have done it now sir

File: server/controller/articleController.js (hunk 11 above)
Comment (goodnessaig1) [reply server/controller/articleController.js]: Ok sir

File: server/controller/articleController.js (hunk 13 above)
Comment (goodnessaig1) [reply server/controller/articleController.js:62]: ok sir. Done

File: server/controller/gifComment.js [hunk 14]
```diff
@@ -11,15 +18,15 @@ class gifController {
       try {
         const { comment } = req.body;
         const { gifId } = req.params;
-        const createdAt = moment().format('YYYY-MM-DD ');
+        const createdAt = getTime();
```
Comment (courage173) [inline server/controller/gifComment.js]: using DateTime.now() is enough for this

File: server/test/gifImg.spec.js (hunk 7 above)
Comment (courage173) [reply server/test/gifImg.spec.js:41]: that means you are not handling the token properly. please set it back to bearer token

**Review by courage173 — CHANGES_REQUESTED** [review]
Getting there. just some things left

**Review by courage173 — APPROVED** [review]
looks good
